src/ZkFold/Base/Protocol/ARK/Plonk.hs

```diff
@@ -113,7 +116,8 @@ newtype PlonkWitnessMap d = PlonkWitnessMap (Map.Map Natural F -> (PolyVec F d,
 newtype PlonkWitnessInput = PlonkWitnessInput (Map.Map Natural F)
 -- TODO (Issue #25): make a proper implementation of Show
 instance Show PlonkWitnessInput where
-    show _ = "PlonkWitnessInput"
+    show (PlonkWitnessInput m) = "Witness Input: " ++ show m
+
 -- TODO (Issue #25): make a proper implementation of Arbitrary
 instance Arbitrary PlonkWitnessInput where
```

Comment:
The problem with this instance is that its definition should be related to the definition of `Arbitrary` for arithmetic circuits. If we allow input variables with indices 1-100 in arbitrary circuits, then this `Map` must contain values for keys 1-100. Keep in mind though that if the circuit's input is constrained to be boolean, then we also need the value to be boolean.